CI: Flaky ResourceWarning: unclosedssl.SSLSocket

https://github.com/pandas-dev/pandas/runs/4399631223?check_suite_focus=true

```
AssertionError: Caused unexpected warning(s): [('ResourceWarning', ResourceWarning("unclosed <ssl.SSLSocket fd=8, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.1.0.137', 51532), raddr=('52.216.164.101', 443)>"), '', 1043)]
```

Initial guess: it may be coming from one of our dependencies like https://github.com/boto/boto3/issues/454

Potential solution:
https://github.com/boto/botocore/pull/1810#issuecomment-749961250
https://github.com/mintel/pytest-localstack/issues/44#issuecomment-848258701

--- a/pandas/_testing/_warnings.py
+++ b/pandas/_testing/_warnings.py
@@ -153,9 +153,8 @@ def _assert_caught_no_extra_warnings(
                 "unclosed transport <asyncio.sslproto._SSLProtocolTransport",
                 "unclosed <ssl.SSLSocket",
             )
-            if (
-                actual_warning.category == ResourceWarning
-                and str(actual_warning.message) in unclosed_ssl
+            if actual_warning.category == ResourceWarning and any(
+                msg in str(actual_warning.message) for msg in unclosed_ssl
             ):
                 # FIXME(GH#38630): kludge because pytest.filterwarnings does not
                 #  suppress these

--- a/pandas/_testing/_warnings.py
+++ b/pandas/_testing/_warnings.py
@@ -147,9 +147,15 @@ def _assert_caught_no_extra_warnings(
 
     for actual_warning in caught_warnings:
         if _is_unexpected_warning(actual_warning, expected_warning):
-            unclosed = "unclosed transport <asyncio.sslproto._SSLProtocolTransport"
-            if actual_warning.category == ResourceWarning and unclosed in str(
-                actual_warning.message
+            # GH 44732: Don't make the CI flaky by filtering SSL-related
+            # ResourceWarning from dependencies
+            unclosed_ssl = (
+                "unclosed transport <asyncio.sslproto._SSLProtocolTransport",
+                "unclosed <ssl.SSLSocket",
+            )
+            if (
+                actual_warning.category == ResourceWarning
+                and str(actual_warning.message) in unclosed_ssl
             ):
                 # FIXME(GH#38630): kludge because pytest.filterwarnings does not
                 #  suppress these

